### ISSUE TITLE
Can't connect inside docker container, "Network is unreachable"

### ISSUE THREAD

**Issue by ziyadrabeh-pki** [sshnet/SSH.NET#1054]
I have a .net 6 app with SSH.NET that works perfectly running outside docker, but if I try to run it inside a docker container I get a "Network is unreachable" exception.

**Comment by ChuckHeatherly** [comment]
@ziyadrabeh-pki or anyone else get any more info on this? I have a workflow that I was testing a lot up until mid-November when I was on leave until two weeks ago, started getting this error when I got back.

My Docker container is a centos-7 base. I'm only getting the "network is unreachable" error when I run it from Docker Desktop on Windows. In production, it runs from a Linux host and runs fine.

Another weird thing that happens - I'll run it and get that error the first time in the program where it creates a new instance of Renci.SshNet.SshClient and calls its Connect() method. Then I'll run it again and it will get past that point but fail at the next instance of that happening. And so on, every new time I run the program, the prior (n-1) connection attempts will succeed but then the next will fail. This is all within the same instance of the container.

Edit: I updated to the latest version of Docker Desktop for Windows, 4.16.3, no changes in behavior. Although there were several network issues fixed in the recent updates, so it still might be a Docker Desktop problem. With enough attempts to run my program, I can eventually get all the connection attempts to succeed, but it seems random as to how many tries it takes.
